### DISCUSSION TITLE
Justify using AS relationship DB

### DISCUSSION THREAD

**Discussion by SichangHe** [SichangHe/internet_route_verification#190] [General]
> - Section 5.1.2, page 5, left, in general) AS relationships are generated from observed BGP paths; hence, there is cyclic dependency when using them to validate the routes they are generated from.

I don't think there is a cyclic dependency. We are matching BGP routes against RPSL, using information from them. AS relationships inferred from BGP paths help make assumptions on special cases, not AS relationships, so there are no cycles.

> Further, the algorithm that generates AS relationships for the CAIDA data set utilizes a large number of heuristics and assumptions---all of which your safelisting implicitly subscribes to when using them.

I don't think this is a problem, @cunha. We explicitly say we use CAIDA's DB, which tells the reader these assumptions.

**Comment by cunha** [comment]
> > * Section 5.1.2, page 5, left, in general) AS relationships are generated from observed BGP paths; hence, there is cyclic dependency when using them to validate the routes they are generated from.
> 
> I don't think there is a cyclic dependency. We are matching BGP routes against RPSL, using information from them. AS relationships inferred from BGP paths help make assumptions on special cases, not AS relationships, so there are no cycles.

It isn't clear to me what the problem is. I think our use of the relationship DB is OK.

I think we should point out that the AS-relationships are also generated from BGP (what the reviewer said). We could mention that our technique doesn't depend on a specific AS-relationship database, any DB would do. Also, the relationship databases usually look at the BGP paths in aggregate, and we look at BGP paths individually.

> > Further, the algorithm that generates AS relationships for the CAIDA data set utilizes a large number of heuristics and assumptions---all of which your safelisting implicitly subscribes to when using them.
> 
> I don't think this is a problem, @cunha. We explicitly say we use CAIDA's DB, which tells the reader these assumptions.

Right; I don't think it's that many assumptions, and they are generally agreed upon (e.g., transit degree, prefer customer > peer > provider).

**Comment by cunha** [comment]
We now mention that any other relationship DB would do, and that errors in the relationship DB may lead to some propagations be misclassified as *relaxed* or *safelisted* when they should instead be *unverified*.

> Our checks rely on the business relationship between each pair of ASes.  We use CAIDA's AS relationship inference database due to its availability and accuracy~\cite{luckie13asrel}.  Incorrectly inferred relationships could result in incorrectly tagging propagation over a link among the \emph{relaxed}, \emph{safelisted}, and \emph{unverified} classifications.
